src/components/HighlightedName/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { useMemo } from 'react';
+import Color from 'color';
+import ColorHash from 'color-hash';
+import memoize from 'lodash/memoize';
+import { Typography, TypographyProps, useTheme } from '@mui/material';
+
+const colorHash = new ColorHash();
+
+export const generateColorFromString = memoize((text: string): string => {
+	console.log('generateColorFromString', text, colorHash);
+	return colorHash.hex(text.replace(/\s/g, ''));
+});
+
+export const isLight = (color?: string) => {
+	return Color(color).luminosity() > 0.65;
+};
+
+export const HighlightedName = ({
+	children,
+	className,
+	color,
+	...props
+}: TypographyProps) => {
+	const theme = useTheme();
+
+	if (typeof children !== 'string') {
+		throw new Error(
+			`The child element of the HighlightedName component must be a string, received: ${typeof children}`,
+		);
+	}
```

Comment:
Wouldn't it be better to omit `children` from `TypographyProps` and add our own `& { children: string }`?